### ISSUE TITLE
Adding more learning rate schedulers

### ISSUE THREAD

**Issue by thomasjpfan** [dnouri/skorch#168]
The `callback` framework in `skorch` allows for more flexible learning rate schedulers. I would like to adapt my PR on **Cyclical Learning Rates**: https://github.com/pytorch/pytorch/pull/2016 to `skorch`. Do you think this would be a good fit for this project?

**Comment by ottonemo** [comment]
Yes, it would definitely fit. We already have a [`WarmRestartLR`](https://github.com/dnouri/skorch/blob/master/skorch/callbacks/lr_scheduler.py#L63) callback which is based on https://arxiv.org/pdf/1608.03983.pdf. While similar, it does not have the ramp-up phase in form of a saw-tooth LR schedule. Wouldn't it make sense to add this as a feature to the warm restart scheduler instead of providing a separate callback?

**Comment by thomasjpfan** [comment]
It seems like the `_LRScheduler ` framework restricts learning rate updates to each epoch. (`_lr_scheduler.step` is called at `on_epoch_begin `). Cyclical Learning Rates is updating the learning rate every batch, meaning it would have to update at `on_batch_begin`. Technically, it is possible to add  a `on_batch_begin` to `WarmRestartLR`, but I think that would cause confusion for a user.

For reference here is the Cyclical Learning Rates paper: https://arxiv.org/abs/1506.01186 and its Keras implementation: https://github.com/bckenstler/CLR.

**Comment by githubnemo** [comment]
Yeah you are right, after comparing both approaches I agree that it makes little sense to combine both approaches in one implementation (implementation differences aside there is also the issue of semantic differences of restarts and a varying LR schedule).

It would be great if you could contribute this as a separate callback to skorch!

**Comment by ottonemo** [comment]
Yeah you are right, after comparing both approaches I agree that it makes little sense to combine both approaches in one implementation (implementation differences aside there is also the issue of semantic differences of restarts and a varying LR schedule).

It would be great if you could contribute this as a separate callback to skorch!
